mkosi/distributions/centos.py

```diff
@@ -38,6 +38,16 @@ def filesystem(cls) -> str:
         # https://marc.info/?l=linux-xfs&m=167450838316386&w=2.
         return "ext4"
 
+    @classmethod
+    def filesystem_options(cls, state: MkosiState) -> dict[str, list[str]]:
+        release = int(state.config.release)
+
+        if release <= 8:
+            # Make the generated ext4 filesystem compatible with the older tooling shipped by CentOS 8 Stream.
+            return dict(ext4=["-O", "^metadata_csum_seed"])
```

Comment:
Also, I do get the less quoting, but 
```suggestion
            return {"ext4"=["-O", "^metadata_csum_seed"]}
```
is just prettier ;)

Reply:
Agree to disagree ;)

Reply:
>  think this will apply more broadly, then CentOS, although I can't think of a good heuristic. It's definitely also needed for Debian up to bullseye.

It probably is, but I'm not interested in older Debian releases, so I'll leave it to someone else to confirm that and submit a PR if needed.

Reply:
/cc @bluca

Reply:
I can check, how do I reproduce it? just build and boot? nspawn or qemu?

Reply:
qemu but you need to hack mkosi to set `SYSTEMD_REPART_MKFS_OPTIONS_EXT4=-O metadata_csum_seed` when calling repart.

Reply:
This is about https://lwn.net/Articles/923969/, so images built on bookworm would probably not be mountable in bullseye (if the change wasn't already reverted to which I'm not entirely clear).